src/context.ts

```diff
@@ -33,15 +33,29 @@ const dbFile = path.join(
   'database.sqlite',
 );
 
-const Context: ContextInterface = {
+interface SessionStorageFactory {
+  SESSION_STORAGE_: SessionStorage | (() => SessionStorage);
+}
+
+const Context: ContextInterface & SessionStorageFactory = {
   API_KEY: '',
   API_SECRET_KEY: '',
   SCOPES: new AuthScopes([]),
   HOST_NAME: '',
   API_VERSION: ApiVersion.Unstable,
   IS_EMBEDDED_APP: true,
   IS_PRIVATE_APP: false,
-  SESSION_STORAGE: new SQLiteSessionStorage(dbFile),
+
+  SESSION_STORAGE_: () => new SQLiteSessionStorage(dbFile),
+  get SESSION_STORAGE(): SessionStorage {
+    if(typeof this.SESSION_STORAGE_ === 'function') {
+      this.SESSION_STORAGE_ = this.SESSION_STORAGE_();
```

Comment:
I wonder if it would make sense to not call the function every time, but call it once on initialization and memoize it. We could let it be an async function, and then the storage class can e.g. connect to a remote DB server when it's created, and we await it here.

Reply:
Right, brilliant!